Neos.FluidAdaptor/Tests/Unit/Core/ViewHelper/AbstractViewHelperTest.php

```diff
@@ -237,14 +237,23 @@ public function validateArgumentsCallsTheRightValidatorsAndThrowsExceptionIfVali
      */
     public function initializeArgumentsAndRenderCallsTheCorrectSequenceOfMethods(): void
     {
+        $calls = [];
         $viewHelper = $this->getAccessibleMock(AbstractViewHelper::class, ['validateArguments', 'initialize', 'callRenderMethod']);
-        $viewHelper->expects(self::at(0))->method('validateArguments');
-        $viewHelper->expects(self::at(1))->method('initialize');
-        $viewHelper->expects(self::at(2))->method('callRenderMethod')->willReturn('Output');
+        $viewHelper->expects(self::atLeastOnce())->method('validateArguments')->willReturnCallback(function () use (&$calls) {
+            $calls[] = 'validateArguments';
+        });
+        $viewHelper->expects(self::atLeastOnce())->method('initialize')->willReturnCallback(function () use (&$calls) {
+            $calls[] = 'initialize';
+        });
+        $viewHelper->expects(self::atLeastOnce())->method('callRenderMethod')->willReturnCallback(function () use (&$calls) {
+            $calls[] = 'callRenderMethod';
+            return 'Output';
+        });
 
         $expectedOutput = 'Output';
         $actualOutput = $viewHelper->initializeArgumentsAndRender(['argument1' => 'value1']);
         self::assertEquals($expectedOutput, $actualOutput);
+        self::assertEquals(['validateArguments', 'initialize', 'callRenderMethod'], $calls);
```

Comment:
Maybe this is just not a good test at all?

Reply:
Could be, but it's named very explicitly: `initializeArgumentsAndRenderCallsTheCorrectSequenceOfMethods()` – so I kept is as such.